packages/hint/docs/contributor-guide/getting-started/release.md

```diff
@@ -0,0 +1,229 @@
+* The script has never been run to its end successfully on Windows and testing changes is complicated.
+* The release can not be fully automated. It uses OTP for GitHub and npm and doesn't take into account other type of authentication mechanisms.
+* Other minor issues like can't continue even if tests fail, auto approve changelog, release an specific package, etc.
+
+### More about circular dependencies
+
+webhint has a circular dependency via `dependencies`, `devDependencies`, `peerDependencies`, and `optionalDependencies`. The circle is as follows (`https-only` is used as an example but it could be any `hint` in `configuration-web-recommended`):
+
+`configuration-web-recommended`
+
+→ `https-only` (`dependencies`)
+
+→ `hint` (`peerDependencies`, `devDependencies`)
+
+→ `configuration-web-recommended` (`optionalDependency`)
+
+Right now it is not clear if the loop is broken somehow. If not, not publishing packages that only have an `Upgrade` in their commit history should be enough. E.g.: If `https-only` gets a `patch`, there will be a commit for `configuration-web-recommended` but nothing should be published so the version will not be bumped.
+
+What should happen when any of these packages gets updated?
+
+If dependency version bumping is done automatically by the release script, everything will be a `patch` (except the package that started the loop that can be anything).
+
+But what if changes are required? **This is something we need to decide**
+Let's use `configuration` as an example. If `configuration` gets a `major` bump. There could be several reasons depending on how we interpret `public API` changes in `semver`:
+
+* A `hint` changes the message reports (`Breaking`). In principle the public API of `config` does not change because it only exposes a `json`, but one could argue that the output has indeed changed. If we align with this, `configuration` should get a major version as well. What happens then to `hint` that depends on `configuration`? Should it get a major bump as well? Because the public API doesn't change, `configuration` is an optional dependency, and it is only used from the CLI, it should be a `patch`. _Right it is not clear what the release script does. Does it assume it is a breaking change in the config? The easiest solution will be to use a `patch`_
+* `configuration` needs to change the `json` format. This means `hint`'s configuration schema has changed as well with another breaking change. It is very unlikely this will work with the current release script knowing that packages are bumped individually and not holistically. `hint` will be the first package to be published and it will point to an old version of `configuration` that is not compatible. Something similar has happened with `hint@4.4.1` that points to `configuration-web-recommended@5.0.0` but the latest version is `6.0.0` (although in this case the configurations are compatible).
```

Comment:
Add case where `hint` is breaking --> `configurations` should break as well.

Reply:
This should go as part of the discussion about configurations so resolving the conversation